core/src/main/java/org/springframework/security/provisioning/JdbcUserPasswordDetailsManager.java

```diff
@@ -0,0 +1,572 @@
+			ps.setString(2, username);
+		});
+		this.userCache.removeUserFromCache(username);
+	}
+
+	@Override
+	public void removeUserFromGroup(final String username, final String groupName) {
+		this.logger.debug("Removing user '" + username + "' to group '" + groupName + "'");
+		Assert.hasText(username, "username should have text");
+		Assert.hasText(groupName, "groupName should have text");
+		int id = findGroupId(groupName);
+		getJdbcTemplate().update(this.deleteGroupMemberSql, (ps) -> {
+			ps.setInt(1, id);
+			ps.setString(2, username);
+		});
+		this.userCache.removeUserFromCache(username);
+	}
+
+	@Override
+	public List<GrantedAuthority> findGroupAuthorities(String groupName) {
+		this.logger.debug("Loading authorities for group '" + groupName + "'");
+		Assert.hasText(groupName, "groupName should have text");
+		return getJdbcTemplate().query(this.groupAuthoritiesSql, new String[] { groupName },
+				this::mapToGrantedAuthority);
+	}
+
+	private GrantedAuthority mapToGrantedAuthority(ResultSet rs, int rowNum) throws SQLException {
+		String roleName = getRolePrefix() + rs.getString(3);
+		return new SimpleGrantedAuthority(roleName);
+	}
+
+	@Override
+	public void removeGroupAuthority(String groupName, final GrantedAuthority authority) {
+		this.logger.debug("Removing authority '" + authority + "' from group '" + groupName + "'");
+		Assert.hasText(groupName, "groupName should have text");
+		Assert.notNull(authority, "authority cannot be null");
+		int id = findGroupId(groupName);
+		getJdbcTemplate().update(this.deleteGroupAuthoritySql, (ps) -> {
+			ps.setInt(1, id);
+			ps.setString(2, authority.getAuthority());
+		});
+	}
+
+	@Override
+	public void addGroupAuthority(final String groupName, final GrantedAuthority authority) {
+		this.logger.debug("Adding authority '" + authority + "' to group '" + groupName + "'");
+		Assert.hasText(groupName, "groupName should have text");
+		Assert.notNull(authority, "authority cannot be null");
+		int id = findGroupId(groupName);
+		getJdbcTemplate().update(this.insertGroupAuthoritySql, (ps) -> {
+			ps.setInt(1, id);
+			ps.setString(2, authority.getAuthority());
+		});
+	}
+
+	private int findGroupId(String group) {
+		return getJdbcTemplate().queryForObject(this.findGroupIdSql, Integer.class, group);
+	}
+
+	/**
+	 * Sets the {@link SecurityContextHolderStrategy} to use. The default action is to use
+	 * the {@link SecurityContextHolderStrategy} stored in {@link SecurityContextHolder}.
+	 *
+	 * @since 5.8
+	 */
+	public void setSecurityContextHolderStrategy(SecurityContextHolderStrategy securityContextHolderStrategy) {
+		Assert.notNull(securityContextHolderStrategy, "securityContextHolderStrategy cannot be null");
+		this.securityContextHolderStrategy = securityContextHolderStrategy;
+	}
+
+	public void setAuthenticationManager(AuthenticationManager authenticationManager) {
+		this.authenticationManager = authenticationManager;
+	}
+
+	public void setCreateUserSql(String createUserSql) {
```

Comment:
I am itching to rename *Sql to *Query since this class now breaks compatibility anyway...